code/modules/mob/living/carbon/human/species/station/skrell/skrell.dm

```diff
@@ -61,8 +61,39 @@
 
 	zombie_type = "Skrell Zombie"
 
+/datum/species/skrell/handle_environment_special(var/mob/living/carbon/human/H)
+	var/old_moisture = H.ST.moisture
+	H.ST.moisture = max(0, H.ST.moisture - rand(-0.8, 1.3))			// Reduce moisture a little.
+	if(H.fire_stacks && H.ST.moisture > 1800)						// Flaming moist skrell can reduce their flame
+		H.fire_stacks--
+		H.ST.moisture -= rand(150, 350)								// Lost a lot of moisture
+	var/pain_limit = (1800-H.ST.moisture) * (40 / 1800)				// The max halloss damage to do based on current moisture. At 0, this is 30. At 1800 or higher, this is 0.
+	
+	// Add more pain damage if they need it.
+	var setPain = max(H.getHalLoss(), pain_limit)
+	H.adjustHalLoss(setPain - H.getHalLoss())
+	
+	// Skrell hydration is directly tied to their moisture.
+	H.hydration = (H.ST.moisture / 3600) * H.max_hydration
+
+	if(H.ST.moisture <= 1800 && old_moisture > 1800)
+		H << "<span class='warning'>You are drying out. You should consider moisturizing.</span>"
+	else if(H.ST.moisture <= 900 && old_moisture > 900)
+		H << "<span class='warning'>You are really dry! Your skin feels uncomfortable and flakey!</span>"
+	else if(H.ST.moisture == 0 && old_moisture > 0)
+		H << "<span class='danger'>You are completely dry, this is really painful!</span>"
```

Comment:
Coding format. Please use to_chat instead of << and SPAN_X("message") instead of span class

Reply:
Fixed by latest patch